src/main/java/com/mongodb/kafka/connect/util/jmx/internal/MBeanServerUtils.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.kafka.connect.util.jmx.internal;
+
+import static java.lang.management.ManagementFactory.getPlatformMBeanServer;
+
+import java.lang.management.ManagementFactory;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.management.AttributeNotFoundException;
+import javax.management.DynamicMBean;
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.IntrospectionException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanException;
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.lang.Nullable;
+
+import com.mongodb.kafka.connect.util.VisibleForTesting;
+
+public final class MBeanServerUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MBeanServerUtils.class);
+
+  @VisibleForTesting(otherwise = VisibleForTesting.AccessModifier.PRIVATE)
+  public static final AtomicInteger NEXT_ID = new AtomicInteger();
+
+  private MBeanServerUtils() {
+    // util class
+  }
+
+  @Nullable
+  public static String registerMBean(final DynamicMBean mBean, final String mBeanName) {
+    MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+    try {
+      String mBeanNameToTry = mBeanName;
+      while (true) {
+        try {
+          server.registerMBean(mBean, new ObjectName(mBeanNameToTry));
+          return mBeanNameToTry;
+        } catch (InstanceAlreadyExistsException e) {
+          LOGGER.warn("MBean name conflict " + mBeanNameToTry, e);
+          mBeanNameToTry = mBeanName + "-v" + NEXT_ID.getAndAdd(1);
+        }
+      }
+    } catch (Exception e) {
+      // JMX might not be available
+      LOGGER.warn("Unable to register MBean " + mBeanName, e);
+      return null;
```

Comment:
Is returning `null` handled / valid here? If I registerMBean and it returns null - will unregistering it cause a NPE instead of being a Noop?

Reply:
unregisterMBean seems to handle unregistering a not-registered MBean.  It both checks for registration (line 83) and swallows exceptions (line 86)

Reply:
Ack - resolving.

Reply:
The logger uses the mbean name, so this would log null in unusual circumstances. I returned an adjusted name.